cellpack/tests/test_db_recipe_handler.py

```diff
@@ -0,0 +1,169 @@
+from cellpack.autopack.DBRecipeHandler import DBRecipeHandler
+from cellpack.tests.mocks.mock_db import MockDB
+
+mock_db = MockDB({})
+
+
+def test_is_nested_list():
+    assert DBRecipeHandler.is_nested_list([]) is False
+    assert DBRecipeHandler.is_nested_list([[], []]) is True
+    assert DBRecipeHandler.is_nested_list([[1, 2], [3, 4]]) is True
+
+
+def test_prep_data_for_db():
+    input_data = {
+        "bounding_box": [[0, 0, 0], [1000, 1000, 1]],
+        "positions": [[(0.0, 0.0, 0.0)], [(0.0, 50.0, 0.0), (43.26, -25.07, 0.0)]],
+        "max_jitter": [1, 1, 0],
+    }
+    converted_data = {
+        "bounding_box": {"0": [0, 0, 0], "1": [1000, 1000, 1]},
```

Comment:
Does firebase not handle numeric keys in dictionaries? I will need a walkthrough of Firebase at some point 😅

Reply:
yeah firebase has some limitations on data types that can be uploaded, we need to convert the data to be "firebase-uploadable" before uploading and revert it back to `cellpack-packable` after downloading. (painful process...)